src/cdk-experimental/menu/focus-mouse-manager.spec.ts

```diff
@@ -0,0 +1,103 @@
+import {Component, QueryList, ElementRef, ViewChildren, AfterViewInit} from '@angular/core';
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
+import {createMouseEvent, dispatchEvent} from '@angular/cdk/testing/private';
+import {FocusMouseManager, FocusableElement} from './focus-mouse-manager';
+
+describe('FocusMouseManger', () => {
+  let fixture: ComponentFixture<MultiElementWithConditionalComponent>;
+  let focusManager: FocusMouseManager<MockWrapper>;
+  let mockElements: MockWrapper[];
+
+  /** Get the components under test from the fixture. */
+  function getComponentsForTesting() {
+    focusManager = fixture.componentInstance.focusManager;
+    mockElements = fixture.componentInstance._allItems.toArray();
+  }
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [MultiElementWithConditionalComponent, MockWrapper],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MultiElementWithConditionalComponent);
+    fixture.detectChanges();
+
+    getComponentsForTesting();
+  });
+
+  it('should emit on mouseEnter observable when mouse enters a tracked element', () => {
+    const spy = jasmine.createSpy('mouse enter spy');
+    focusManager.mouseEntered.subscribe(spy);
+
+    const event = createMouseEvent('mouseenter');
+    dispatchEvent(mockElements[0]._elementRef.nativeElement, event);
+    fixture.detectChanges();
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenCalledWith(mockElements[0]);
+  });
+
+  it('should be aware of newly created/added components and track them', () => {
+    const spy = jasmine.createSpy('mouse enter spy');
+    focusManager.mouseEntered.subscribe(spy);
+
+    expect(fixture.componentInstance.showThird).toBeFalse();
```

Comment:
I'm not sure this adds anything, since it's clear in the test host setup that this starts out false. (Also, it doesn't look like it's being updated in response to anything under test.)

Reply:
It's more of a sanity check before the test, but I see how that's not clear. I'll make a change to make that more obvious.

The functionality that's being tested here is that when a new element is added, the focus-manager updates and picks it up.